core/src/main/scala/org/http4s/internal/package.scala

```diff
@@ -117,4 +119,14 @@ package object internal {
           }
       }
     }
+
+  private[http4s] def allocated[F[_], A](resource: Resource[F, A])(
```

Comment:
This is too useful, whats the catch? The stronger constraint?

Reply:
That's the only downside I can see.